packages/legend/lib/create-legend/from-layer.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  MapContextLayer,
+  MapContextLayerWms,
+  MapContextLayerWmts,
+  removeSearchParams,
+} from "@geospatial-sdk/core";
+import { WmtsEndpoint } from "@camptocamp/ogc-client";
+
+/**
+ * Configuration options for legend generation
+ */
+interface LegendOptions {
+  format?: string;
+  widthPxHint?: number;
+  heightPxHint?: number;
+}
+
+/**
+ * Create a legend URL for a WMS layer
+ *
+ * @param layer - The MapContextLayer to create a legend URL for
+ * @param options - Optional configuration for legend generation
+ * @returns A URL for the WMS legend graphic
+ */
+function createWmsLegendUrl(
+  layer: MapContextLayerWms,
+  options: LegendOptions = {},
+): URL {
+  const { format = "image/png", widthPxHint, heightPxHint } = options;
+
+  const legendUrl = new URL(
+    removeSearchParams(layer.url, [
+      "SERVICE",
+      "REQUEST",
+      "FORMAT",
+      "LAYER",
+      "LAYERTITLE",
+      "WIDTH",
+      "HEIGHT",
+    ]),
+  );
+  legendUrl.searchParams.set("SERVICE", "WMS");
+  legendUrl.searchParams.set("REQUEST", "GetLegendGraphic");
+  legendUrl.searchParams.set("FORMAT", format);
+  legendUrl.searchParams.set("LAYER", layer.name);
+  legendUrl.searchParams.set("LAYERTITLE", false.toString()); // Disable layer title for QGIS Server
+
+  if (widthPxHint) {
+    legendUrl.searchParams.set("WIDTH", widthPxHint.toString());
+  }
+  if (heightPxHint) {
+    legendUrl.searchParams.set("HEIGHT", heightPxHint.toString());
+  }
+
+  return legendUrl;
+}
+
+/**
+ * Create a legend URL for a WMTS layer
+ *
+ * @param layer - The MapContextLayer to create a legend URL for
+ * @returns A URL for the WMTS legend graphic or null if not available
+ */
+async function createWmtsLegendUrl(
+  layer: MapContextLayerWmts,
+): Promise<string | null> {
+  const endpoint = await new WmtsEndpoint(layer.url).isReady();
+
+  const layerByName = endpoint.getLayerByName(layer.name);
+  console.log("layerByName");
+  console.log(layerByName);
+
+  if (
+    layerByName.styles &&
+    layerByName.styles.length > 0 &&
+    layerByName.styles[0].legendUrl
+  ) {
+    return layerByName.styles[0].legendUrl;
+  }
+
+  return null;
+}
+
+/**
+ * Create a legend from a layer
+ *
+ *
+ * @param layer - The MapContextLayer to create a legend from
+ * @param options - Optional configuration for legend generation
+ * @returns The legend as a DOM element or false if the legend could not be created
+ */
+export async function createLegendFromLayer(
+  layer: MapContextLayer,
+  options: LegendOptions = {},
+): Promise<HTMLElement | false> {
```

Comment:
```suggestion
): Promise<HTMLElement | null> {
```

It is more common to return `null` when something could no be found/created.